mogp_emulator/tests/test_Priors.py

```diff
@@ -1,48 +1,822 @@
+    theta = GPParams(n_mean=1, n_corr=1, fit_cov=True, nugget="fit")
+    theta.set_data(np.zeros(3))
+    
+    logp = gpp.logp(theta)
+    
+    assert_allclose(logp, np.sum([ float(LogNormalPrior(2., 3.).logp(theta.corr)),
+                                   float(GammaPrior(2., 3.).logp(theta.cov)),
+                                   float(InvGammaPrior(1., 1.).logp(theta.nugget)) ]))
+                                   
+    # gpp = GPPriors(corr=[ LogNormalPrior(2., 3.)], cov=InvGammaPrior(2., 3.), nugget=InvGammaPrior(1., 1.),
+    #                nugget_type="fit")
+    #
+    # theta = GPParams(n_mean=1, n_corr=1, fit_cov=False, nugget="fit")
+    # theta.set_data(np.zeros(2))
+    #
+    # logp = gpp.logp(theta)
+    #
+    # assert_allclose(logp, np.sum([ float(LogNormalPrior(2., 3.).logp(theta.corr)),
+    #                                float(InvGammaPrior(1., 1.).logp(theta.nugget)) ]))
+
+    gpp = GPPriors(corr=[ LogNormalPrior(2., 3.)], cov=GammaPrior(2., 3.),
+                   nugget_type="pivot")
+                   
+    theta = GPParams(n_mean=1, n_corr=1, fit_cov=True, nugget="pivot")
+    theta.set_data(np.zeros(2))
+    
+    logp = gpp.logp(theta)
+    
+    assert_allclose(logp, np.sum([ float(LogNormalPrior(2., 3.).logp(theta.corr)),
+                                   float(GammaPrior(2., 3.).logp(theta.cov)) ]))
+
+    with pytest.raises(TypeError):
+        gpp.logp(1.)
+
+    theta = GPParams(n_mean=1, n_corr=2, fit_cov=True, nugget="pivot")
+    
+    with pytest.raises(AssertionError):
+        gpp.logp(theta)
+        
+    # theta = GPParams(n_mean=1, n_corr=1, fit_cov=False, nugget="pivot")
+    #
+    # with pytest.raises(AssertionError):
+    #     gpp.logp(theta)
+        
+    theta = GPParams(n_mean=1, n_corr=1, fit_cov=True, nugget="fit")
+    
+    with pytest.raises(AssertionError):
+        gpp.logp(theta)
+        
+    theta = GPParams(n_mean=1, n_corr=1, fit_cov=True, nugget="pivot")
+    
+    with pytest.raises(AssertionError):
+        gpp.logp(theta)
+                      
+def test_GPPriors_dlogpdtheta():
+    "test the dlogpdtheta method of the GPPriors class"
+    
+    gpp = GPPriors(corr=[ LogNormalPrior(2., 3.)], cov=GammaPrior(2., 3.), nugget=InvGammaPrior(1., 1.),
+                   nugget_type="fit")
+                    
+    theta = GPParams(n_mean=1, n_corr=1, fit_cov=True, nugget="fit")
+    theta.set_data(np.zeros(3))
+    
+    partials = gpp.dlogpdtheta(theta)
+    
+    assert_allclose(partials,
+                    [ float(LogNormalPrior(2., 3.).dlogpdx(theta.corr)*CorrTransform.dscaled_draw(theta.corr)),
+                      float(GammaPrior(2., 3.).dlogpdx(theta.cov)*CovTransform.dscaled_draw(theta.cov)),
+                      float(InvGammaPrior(1., 1.).dlogpdx(theta.nugget)*CovTransform.dscaled_draw(theta.nugget))])
+                      
+    # gpp = GPPriors(corr=[ LogNormalPrior(2., 3.)], cov=InvGammaPrior(2., 3.), nugget=InvGammaPrior(1., 1.),
```

Comment:
I guess this commented-out stuff could be removed at some point..

Reply:
Actually, these are tests for functionality that I ended up not implementing but will need later. They will hang around until I finish the other new feature and then be resurrected.